### PR TITLE
Fix Azure SP Personal Access Token Duration

### LINKED ISSUE #66: [ISSUE] Databricks provider uses invalid access token
### Terraform Version
```
➜ terraform -v
Terraform v0.12.25
+ provider.azuread v0.9.0
+ provider.azurerm v2.11.0
+ provider.databricks (unversioned)
+ provider.http v1.2.0
+ provider.null v2.1.2
+ provider.random v2.2.1
+ provider.time v0.5.0
```
Current master branch

### Affected Resource(s)
- all resources

### Terraform Configuration Files
https://github.com/datarootsio/terraform-module-azure-datalake/blob/a3c400b5bf40c2d64159bd703428c062f0174a23/databricks.tf

### Debug Output
https://github.com/datarootsio/terraform-module-azure-datalake/runs/709963745?check_suite_focus=true

```
2020-05-26T16:40:31.5532525Z     command.go:172: Error: Response from server (403) <html>
2020-05-26T16:40:31.5532687Z     command.go:172: <head>
2020-05-26T16:40:31.5533058Z     command.go:172: <meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
2020-05-26T16:40:31.5533244Z     command.go:172: <title>Error 403 Invalid access token.</title>
2020-05-26T16:40:31.5533404Z     command.go:172: </head>
2020-05-26T16:40:31.5533565Z     command.go:172: <body><h2>HTTP ERROR 403</h2>
2020-05-26T16:40:31.5533748Z     command.go:172: <p>Problem accessing /api/2.0/secrets/put. Reason:
2020-05-26T16:40:31.5533922Z     command.go:172: <pre>    Invalid access token.</pre></p>
2020-05-26T16:40:31.5534079Z     command.go:172: </body>
2020-05-26T16:40:31.5534228Z     command.go:172: </html>
2020-05-26T16:40:31.5534573Z     command.go:172: : invalid character '<' looking for beginning of value
2020-05-26T16:40:31.5534739Z     command.go:172: 
2020-05-26T16:40:31.5534912Z     command.go:172:   on databricks.tf line 73, in resource "databricks_secret" "cmdb_master":
2020-05-26T16:40:31.5535168Z     command.go:172:   73: resource "databricks_secret" "cmdb_master" {
```

### Expected Behavior
Create a `databricks_secret`

### Actual Behavior
See error output above

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Use the module https://github.com/datarootsio/terraform-module-azure-datalake as described in the readme

### Comments
At first sight I thought it was an issue with `databricks_token` but it does not seem to be directly related to that resource. It seems to be an issue with the token that this provider is using underneath to create the resources as this seems to happen with a `databricks_secret` that isn't using any access tokens explictly.

### PR DESCRIPTION
**Pull request by stikkireddy**
This lets the user control the duration of PAT token that the user creates. it introduces a new field 
`pat_token_duration_seconds` to the `azure_auth` block in the provider.

- Changed the token create client to accept time.Duration rather than int32
- Made the pat token duration a field in the azure_auth block but will be deprecated for preference of the AAD token when it fully supports all the Databricks APIs
- Added DBApiClientMetadata to the DBApiClientConfig to help with testing  
- Added docs for the field `pat_token_duration_seconds`

Fixes #66

### CODE REVIEW

**Comment by stikkireddy** [comment]
@sdebruyn this should resolve your issue and let the duration be customizable please take a quick look at this and if you have any comments on the new field name please let me know.

**Comment by sdebruyn** [comment]
LGTM!

File: client/service/client.go [hunk 1]
```diff
@@ -59,6 +59,11 @@ const (
 
 var clientAuthorizerMutex sync.Mutex
 
+// DBApiClientMetadata used for unit testing and verifying client behavior
+type DBApiClientMetadata struct {
+	AzureAuthPatTokenExpiry int32
```
Comment (nfx) [inline client/service/client.go:64]: why do we heed a struct just with single field? looks like anti-pattern

File: databricks/azure_auth.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ type TokenPayload struct {
 	ClientSecret         string
 	ClientID             string
 	TenantID             string
+	PatTokenDuration     int32
```
Comment (nfx) [inline databricks/azure_auth.go:39]: this does not logically belong to azure auth, it's a mix of concerns here

File: client/service/client.go (hunk 1 above)
Comment (stikkireddy) [reply client/service/client.go]: noted and removed 😄

**Pull request merged**